### PR TITLE
Add password policy endpoint in auth (#1)

### PR DESCRIPTION
**Pull request by ch5zzy**
Add an endpoint for fetching the password policy in auth, as this will be used to implement password validation against project/tenant policies.

### CODE REVIEW

File: packages/auth/src/api/password_policy/get_password_policy.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    auth = await testAuth();
+    mockFetch.setUp();
+  });
+
+  afterEach(mockFetch.tearDown);
+
+  it('should GET to the correct endpoint', async () => {
+    const mock = mockEndpoint(Endpoint.GET_PASSWORD_POLICY, {
+      customStrengthOptions: {
+        minPasswordLength: 6
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/api/password_policy/get_password_policy.test.ts:44]: Can you put all test data in magic constant?  e.g.  `const TEST_MIN_PASSWORD_LENGTH = 6;`

File: packages/auth/src/api/password_policy/get_password_policy.test.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    expect(response.customStrengthOptions.minPasswordLength).to.eql(6);
+    expect(response.allowedNonAlphanumericCharacters).to.eql(['!']);
+    expect(response.schemaVersion).to.eql(1);
+    expect(mock.calls[0].method).to.eq('GET');
+    expect(mock.calls[0].headers!.get(HttpHeader.X_CLIENT_VERSION)).to.eq(
+      'testSDK/0.0.0'
+    );
+  });
+
+  it('should handle errors', async () => {
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/api/password_policy/get_password_policy.test.ts:60]: maybe rewording to should handle invalid provider id errors?

Optional - Also, are there any other errors will be thrown here? Maybe we can add more tests cases after we raise the error handling PR?

File: packages/auth/src/api/password_policy/get_password_policy.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  HttpMethod,
+  _addTidIfNecessary
+} from '../index';
+import { Auth } from '../../model/public_types';
+
+export interface GetPasswordPolicyRequest {
+  tenantId?: string;
+}
+
+export interface GetPasswordPolicyResponse {
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/api/password_policy/get_password_policy.ts:30]: Can you add reference doc here using /** ... */ ? Same for line 43 and line 26.

**Review by Xiaoshouzi-gh — CHANGES_REQUESTED** [review]

File: packages/auth/src/api/password_policy/get_password_policy.test.ts (hunk 2 above)
Comment (ch5zzy) [reply packages/auth/src/api/password_policy/get_password_policy.test.ts:60]: For this test, I was following the convention for the other endpoint test cases. It seems if there is no specific error associated with the endpoint, they test this error. There are other server errors that can happen but none that are specific to fetching the password policy.

File: packages/auth/src/api/password_policy/get_password_policy.ts (hunk 3 above)
Comment (ch5zzy) [reply packages/auth/src/api/password_policy/get_password_policy.ts:30]: The other files under the directories in src/api didn't have reference docs so I wasn't sure if I was supposed to add them.

Comment (ch5zzy) [reply packages/auth/src/api/password_policy/get_password_policy.ts:36]: Done.

File: packages/auth/src/api/password_policy/get_password_policy.test.ts (hunk 1 above)
Comment (ch5zzy) [reply packages/auth/src/api/password_policy/get_password_policy.test.ts]: Done.

**Review by renkelvin — APPROVED** [review]
LGTM.

**Review by Xiaoshouzi-gh — APPROVED** [review]
LGTM
